### PR TITLE
Stringly Typed Enums

### LINKED ISSUE #7: SDLJingle 'constants' should be constants.
currently defined as

```
+(NSString*) NEGATIVE_JINGLE { return @"NEGATIVE_JINGLE"; }
+(NSString*) POSITIVE_JINGLE { return @"POSITIVE_JINGLE"; }
+(NSString*) LISTEN_JINGLE { return @"LISTEN_JINGLE"; }
+(NSString*) INITIAL_JINGLE { return @"INITIAL_JINGLE"; }
+(NSString*) HELP_JINGLE { return @"HELP_JINGLE"; }
```

see #3 for more information

### LINKED ISSUE #425: Xcode 8 adds Stringly Typed Obj-C Constant Conversion for Swift
Covers [Proposal SDL-0006](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0006-ios-stringly-typed-enums.md)

When Xcode 8 releases, we will have the ability to import Obj-C string constants as Enums using `NS_EXTENSIBLE_STRING_ENUM`.

### PR DESCRIPTION
**Pull request by asm09fsu**
Fixes #425 
Fixes #7 
Proposal [SDL-0006](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0006-ios-stringly-typed-enums.md).

This PR is **not ready** for review.

### Risk
This PR makes **major** API changes.

### Testing Plan
This will be tested using the Specs written in SDL.

### Summary
This PR includes changes to SDL to replace the `SDLEnum` class with stringly typed enums of a `NSString*` typedef'd as `SDLEnum`.

##### Breaking Changes
* `SDLEnum` class remove in support for `SDLEnum` typedef constants.

### Tasks Remaining:
- [x] Update SmartDeviceLink
- [x] Update SmartDeviceLink-Example
- [x] Update SmartDeviceLinkTests
- [x] Validate Naming for all SDLEnums to confirm good readability.

### CODE REVIEW

**Comment by asm09fsu** [comment]
@joeljfischer review names for all enums when you can. We need to make sure they are as readable as can be.

File: README.md [hunk 1]
```diff
@@ -198,7 +198,7 @@ When your app receives `onOnHMIStatus` it has changed HMI states on the head uni
 You will want to track your first HMI FULL, for instance with a boolean value. The example application has extremely basic tracking of this type in the `onOnHMIStatus` callback.
 
 ```objc
-if ((notification.hmiLevel == [SDLHMILevel FULL]) && self.isFirstHMIFull) {
```
Comment (joeljfischer) [inline README.md]: The README should change quite a bit for 4.3, but I think this would have to use `isEqualToString:`?

**Comment by asm09fsu** [comment]
@joeljfischer please review Enum names to confirm that the readability is as good as can be in both ObjC and Swift.

**Comment by joeljfischer** [comment]
@asm09fsu Let me know what you think of these candidates:
`SDLResultCharLimitExceeded` -> `SDLResultCharacterLimitExceeded`
`SDLTriggerSourceVR` -> `SDLTriggerSourceVoice` (I don't know if any other enums with `VR` would also be a candidate for change?)

**Comment by asm09fsu** [comment]
@joeljfischer review them now.

**Pull request merged**